ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -179,12 +179,19 @@ export function AssetPickerModal({
 
     function* tokenGenerator() {
       yield nativeToken;
+
       for (const token of memoizedUsersTokens) {
         yield token;
       }
-      for (const token of Object.values(topTokens)) {
-        yield token;
+
+      // topTokens should already be sorted by popularity
+      for (const address of Object.keys(topTokens)) {
+        const token = tokenList?.[address];
```

Comment:
Do the cases of topTokens and tokenList addresses always match?

Reply:
`topTokens` doesn't map to tokens but rather to the position of the token (e.g., `{'index': 2}`)

Reply:
<img width="349" alt="Screenshot 2024-05-14 at 2 50 22 PM" src="https://github.com/MetaMask/metamask-extension/assets/44588480/19edaf0e-06a9-4e32-99c7-66b4beaa7f4f">

Reply:
I meant the cases of their keys. Looks like they both use lowercase addresses